src/commands/Minion/open.ts

```diff
@@ -142,36 +158,76 @@ export default class extends BotCommand {
-		const openable = Openables.find(thing =>
-			thing.aliases.some(alias => stringMatches(alias, type))
+	async osjsOpenablesOpen(msg: KlasaMessage, quantity: number, osjsOpenable: any) {
+		const userBank = msg.author.settings.get(UserSettings.Bank);
+		if ((userBank[osjsOpenable.id] ?? 0) < quantity) {
+			throw `You don't have enough ${
+				osjsOpenable.name
+			} to open!\n\n However... ${await this.showAvailable(msg)}`;
+		}
+
+		await msg.author.removeItemFromBank(osjsOpenable.id, quantity);
+
+		const loot = osjsOpenable.open(quantity);
+
+		this.client.emit(
+			Events.Log,
+			`${msg.author.username}[${msg.author.id}] opened ${quantity} ${osjsOpenable.name}.`
 		);
 
-		if (!openable) {
+		const task = this.client.tasks.get('bankImage')!;
+		const image = await task.generateBankImage(
+			loot,
+			`You opened ${quantity} ${osjsOpenable.name}`,
+			true,
+			{ showNewCL: 1 },
+			msg.author
+		);
+
+		await msg.author.addItemsToBank(loot, true);
+
+		return msg.send(new MessageAttachment(image, 'osbot.png'));
+	}
+
+	async botOpenablesOpen(msg: KlasaMessage, quantity: number, name: string) {
+		const botOpenable = botOpenables.find(thing =>
+			thing.aliases.some(alias => stringMatches(alias, name))
+		);
+
+		if (!botOpenable) {
 			throw `That's not a valid thing you can open. You can open a clue tier (${ClueTiers.map(
 				tier => tier.name
-			).join(', ')}), or another non-clue thing (${Openables.map(thing => thing.name).join(
-				', '
-			)})`;
+			).join(', ')}), or another non-clue thing (${botOpenables
+				.map(thing => thing.name)
+				.concat(Openables.map(thing => thing.name))
+				.join(', ')})`;
 		}
 
-		const hasItem = await msg.author.hasItem(openable.itemID);
-		if (!hasItem) {
-			throw `You don't have a ${itemNameFromID(openable.itemID)} to open!`;
+		const userBank = msg.author.settings.get(UserSettings.Bank);
+		if ((userBank[botOpenable.itemID] ?? 0) < quantity) {
+			throw `You don't have enough ${
+				botOpenable.name
+			} to open!\n\n However... ${await this.showAvailable(msg)}`;
 		}
 
-		await msg.author.removeItemFromBank(openable.itemID);
+		await msg.author.removeItemFromBank(botOpenable.itemID, quantity);
 
-		const loot = bankFromLootTableOutput(openable.table.roll());
+		const loot = new Loot();
+		for (let i = 0; i < quantity; i++) {
+			loot.add(botOpenable.table.roll());
+		}
 
-		await msg.author.addItemsToBank(loot, true);
+		await msg.author.addItemsToBank(loot.values(), true);
 
-		return msg.send(
-			`${openable.emoji} You opened a ${openable.name} and received: ${
-				Object.keys(loot).length > 0
-					? await createReadableItemListFromBank(this.client, loot)
-					: 'Nothing! Sad'
-			}.`
+		const task = this.client.tasks.get('bankImage')!;
+		const image = await task.generateBankImage(
+			loot.values(),
+			`You opened ${quantity} ${botOpenable.name}`,
+			true,
+			{ showNewCL: 1 },
+			msg.author
 		);
+
+		return msg.send(new MessageAttachment(image, 'osbot.png'));
```

Comment:
use `return msg.channel.sendBankImage` instead of these 3 things, you can find examples of using it in the bank cmd: https://github.com/oldschoolgg/oldschoolbot/blob/master/src/commands/Minion/bank.ts

(and to anywhere else in this file that does this)